src/simple_server/core.clj

```diff
@@ -14,21 +14,68 @@
+                 "Or go to /log-in?account=\"your account\" to log in to start playing the game!")))
+
+(defn sign-up-handler [account]
+  ;; Check if the user already exists, if not, create it, otherwise, tell the user to log in
+  (if (nil? (some #{account} @users))
+    (do
+      (swap! users conj account)
+      (response (str "Thank you for joining us! " account ". You may now log in to your page at /login?account=" account)))
+    (response (str  "You are already with us " account ", log in to your page at /login?account=" account))))
+
+(defn log-in-handler [account]
+  ;; Check if the user is in our database, if not, tell the user to sign up, otherwise, log the user in
+  (if (nil? (some #{account} @users))
+    (response "You are not a member of us yet, pleaes sign up at /sign-up?account=\"your account\".")
+    (do
+      (swap! current-users conj account)
+      (response (str "Welcome back " account ", head to your page at /" account)))))
+
+(defn account-handler [account]
+  ;; Check if the user is logged in, if not, tell the user to log in, otherwise, welcome the user
+  (if (nil? (some #{account} @current-users))
+    (response "Please first log in via /log-in?account=\"your account\"")
+    (response (str "Hello " account ". Start your game at /" account "/new-game"))))
+
+(defn new-game-handler [account]
+  ;; Check if the user is logged in, if not, tell the user to log in, otherwise, instruct the user to start the game
+  (if (nil? (some #{account} @current-users))
+    (response "You must be logged-in to access your account. Please log-in at /log-in?account=" account)
+    (do
+      (when (game/new-game!)
+        (response (str "OK- start guessing at /" account "/guess?guess=x."))))))
+
+(defn guess-handler [guess account]
+  ;; Players must be a logged in user in order to play the game
+  (if (nil? (some #{account} @current-users))
+    (response "You must be logged-in before playing the game. Please head to /log-in?account=\"your account\".")
+    (do
+      (condp = (game/guess-answer guess)
+        nil       (-> (response  "You need to supply a guess with /guess?guess=N")
+                      (status 400))
+        :you-win   (response (str "Congratulations! You win " account "! Play again at /" account "/new-game"))
+        :you-lose  (response (str "Sorry, you ran out of guesses " account ". It's GAME OVER! Play again at /" account "/new-game"))
+        :too-low   (response (str "Too low " account ", " (get-in @game/game-in-progress [:guesses]) " guesses left."))
+        :too-high  (response (str "Too high " account ", " (get-in @game/game-in-progress [:guesses]) " guesses left."))))))
 
 (defroutes game-routes
-  (GET "/new-game" []                 (new-game-handler))
-  (GET "/guess"    [guess :<< as-int] (guess-handler guess))
+  (GET "/landing"  [account]          (assoc-in (landing-handler) [:headers "Content-type"] "text/html"))  
```

Comment:
I'm sure you realise this duplication of assoc [:headers ..] is nonsense, but you probably just didn't figure out how to get middleware to work.

The other unfortunate duplication is how several of your handlers need to check if you are logged in; this could also be centralized in a middleware.

We'll see how to do this in class, OK?

Reply:
Yes please..